src/ast/seq_decl_plugin.h

```diff
@@ -559,18 +579,48 @@ class seq_util {
         app* mk_epsilon(sort* seq_sort);
         info get_info(expr* r) const;
         std::string to_str(expr* r) const;
+        std::string to_strh(expr* r) const;
+
+        expr_ref mk_ite_simplify(expr* c, expr* t, expr* e)
```

Comment:
use bool_rewriter

Reply:
The issue here is that `bool_rewriter` may get rid of ite structure, while where `mk_ite_simplify` is being used it is important to keep the if-then-else structure in the **transition-regexes**. So here, I'm keeping it for now.